### PR TITLE
Use caches instead of get_cache

### PR DESCRIPTION
**Pull request by dannyroberts**
need to merge and deploy https://github.com/dimagi/commcarehq-ansible/pull/211 first

### CODE REVIEW

**Comment by czue** [comment]
build fails look real.
```
Traceback (most recent call last):
  File "manage.py", line 66, in <module>
    execute_from_command_line(sys.argv)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 385, in execute_from_command_line
    utility.execute()
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 377, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/commands/test.py", line 50, in run_from_argv
    super(Command, self).run_from_argv(argv)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 288, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/commands/test.py", line 71, in execute
    super(Command, self).execute(*args, **options)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 338, in execute
    output = self.handle(*args, **options)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/commands/test.py", line 88, in handle
    failures = test_runner.run_tests(test_labels)
  File "/home/travis/build/dimagi/commcare-hq/testrunner.py", line 224, in run_tests
    full_suite = self.build_suite(test_labels, extra_tests)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/test/simple.py", line 242, in build_suite
    suite.addTest(build_suite(app_config))
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/test/simple.py", line 144, in build_suite
    tests_module = get_tests(app_config)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/test/simple.py", line 102, in get_tests
    test_module = import_module('%s.%s' % (app_config.name, TEST_MODULE))
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/callcenter/tests/__init__.py", line 2, in <module>
    from corehq.apps.callcenter.tests.test_indicators import *
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/callcenter/tests/test_indicators.py", line 18, in <module>
    locmem_cache = cache.caches['django.core.cache.backends.locmem.LocMemCache']
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/cache/__init__.py", line 110, in __getitem__
    "Could not find config for '%s' in settings.CACHES" % alias
InvalidCacheBackendError: Could not find config for 'django.core.cache.backends.locmem.LocMemCache' in settings.CACHES
```

File: corehq/apps/callcenter/tests/test_indicators.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 CASE_TYPE = 'cc_flw'
 
-locmem_cache = cache.get_cache('django.core.cache.backends.locmem.LocMemCache')
+locmem_cache = cache.caches['django.core.cache.backends.locmem.LocMemCache']
```
Comment (snopoke) [inline corehq/apps/callcenter/tests/test_indicators.py]: I think this should be `caches['locmem']`

File: corehq/apps/hqadmin/views.py [hunk 2]
```diff
@@ -866,7 +866,7 @@ def view_data(case_id, indicators):
         }
 
     if user or user_case:
-        custom_cache = None if enable_caching else cache.get_cache('django.core.cache.backends.dummy.DummyCache')
+        custom_cache = None if enable_caching else cache.caches['django.core.cache.backends.dummy.DummyCache']
```
Comment (snopoke) [inline corehq/apps/hqadmin/views.py]: I think you'll have to add this to settings as well.

**Comment by dannyroberts** [comment]
(Also https://github.com/dimagi/dimagi-utils/pull/317 though it's technically not dependent I think)

**Comment by dannyroberts** [comment]
Figured it was easier/safer/more logical to add locmem and dummy aliases to CACHES directly in settings, rather than asking everyone to add it to their localsettings. Unlike redis, there's no machine-dependent configuration necessary.

File: settings.py [hunk 3]
```diff
@@ -1490,3 +1490,9 @@
 }
 
 COMPRESS_CSS_HASHING_METHOD = 'content'
+
+
+if 'locmem' not in CACHES:
+    CACHES['locmem'] = {'BACKEND': 'django.core.cache.backends.locmem.LocMemCache'}
+if 'dummy' not in CACHES:
+    CACHES['dummy'] = {'BACKEND': 'django.core.cache.backends.dummy.DummyCache'}
```
Comment (czue) [inline settings.py]: this seems a little sneaky but can't think of any reason why it would break anything or we shouldn't do it.

**Comment by dannyroberts** [comment]
:white_check_mark:

**Pull request merged**